gnovm/pkg/gnolang/type_check.go

```diff
@@ -734,7 +737,13 @@ func (x *BinaryExpr) AssertCompatible(lt, rt Type) {
 		case EQL, NEQ:
 			assertComparable(xt, dt)
 			if !isUntyped(xt) && !isUntyped(dt) {
-				assertAssignableTo(x, xt, dt)
+				err := checkAssignableTo(x, xt, dt)
+				if err != nil {
+					if debug {
+						debug.Printf("checkAssignableTo fail: %v\n", err)
+					}
+					panic(fmt.Sprintf("invalid operation: %v (mismatched types %v and %v)", x, xt, dt))
```

Comment:
security-wise, we should maybe avoid panicking with the whole expression, as it might be very deep. what about making the error "types %v and %v are not comparable"?